contracts/Market.sol

```diff
@@ -1267,24 +1297,29 @@ contract Market is
     });
     bool isTransferSuccessful;
     uint8 holdbackPercentage;
-    uint256 restrictedSupplierFee;
-    uint256 unrestrictedSupplierFee;
+    uint256 restrictedSupplierFee; // purchasing token units
+    uint256 unrestrictedSupplierFee; // purchasing token units
     for (uint256 i = 0; i < countOfRemovalsAllocated; ++i) {
       holdbackPercentage = _removal.getHoldbackPercentage({id: removalIds[i]});
 
-      unrestrictedSupplierFee = removalAmounts[i].mulDiv(_priceMultiple, 100);
+      unrestrictedSupplierFee = convertRemovalAmountToPurchasingTokenAmount(
+        removalAmounts[i].mulDiv(_priceMultiple, 100)
+      );
       if (holdbackPercentage > 0) {
-        restrictedSupplierFee = removalAmounts[i].mulDiv(
-          _priceMultiple * holdbackPercentage,
-          10000
+        restrictedSupplierFee = convertRemovalAmountToPurchasingTokenAmount(
+          removalAmounts[i].mulDiv(_priceMultiple * holdbackPercentage, 10000)
         );
         unrestrictedSupplierFee -= restrictedSupplierFee;
         if (
           _restrictedNORI.getUnderlyingTokenAddress() !=
           address(_purchasingToken)
         ) {
           emit SkipRestrictedNORIERC20Transfer({
-            amount: restrictedSupplierFee,
+            // This assumes that rnori will only ever be backed by NORI or a 1:1 18 decimal token ...
+            // Not ideal to have to convert back to Removal units here but it minimizes the code changes elsewhere.
+            amount: convertPurchasingTokenAmountToRemovalAmount(
+              restrictedSupplierFee
+            ),
```

Comment:
@amiecorso I realized I'm not clear on what the units are supposed to be for this amount parameter.  I've configured it to be removals at the moment but I guess since the purchasing token is an argument it could just as well be the purchase token amount. 

The docs just say `The amount of RestrictedNORI in the transfer attempt.` which is ambiguous in this fun new world.  :/

Reply:
so the `restrictedSupplierFee` and `unrestrictedSupplierFee` are currency amounts, not removal amounts, since the RestrictedNORI contract handles currency and these values are generated from removal amounts while incorporating the price multiple.   I'm thinking that RestrictedNORI will work fine as-is, assuming the underlying token that it wraps is never changed after schedules are already in action (something that wouldn't make sense anyway, even if there wasn't this additional precision specter).  That's because it's always working 1:1 with amounts of underlying token and amounts of RestrictedNORI that get minted to represent that underlying token.  So when going to make a withdrawal, if there was a schedule holding $1 there would have been 1_000_000 RestrictedNORI minted there in the first place.  

So this is correct to apply decimal-related conversions to any currency amounts being passed to RestrictedNORI functions.

Reply:
Ok, so amount in the error event is in _purchasingToken currency units.  I will remove the change I made then.

Reply:
yeah should always be in currency, everything about restrictedNORI 👍

Reply:
Reverted my change to this event so amount is in _paymentToken units.